src/coreclr/vm/jithelpers.cpp

```diff
@@ -1382,24 +1382,6 @@ HCIMPL2(Object*, JIT_Box, CORINFO_CLASS_HANDLE type, void* unboxedData)
 }
 HCIMPLEND
 
-/*************************************************************/
-HCIMPL2_IV(LPVOID, JIT_GetRefAny, CORINFO_CLASS_HANDLE type, TypedByRef typedByRef)
-{
-    FCALL_CONTRACT;
-
-    TypeHandle clsHnd(type);
-    // <TODO>@TODO right now we check for precisely the correct type.
-    // do we want to allow inheritance?  (watch out since value
-    // classes inherit from object but do not normal object layout).</TODO>
-    if (clsHnd != typedByRef.type) {
-        FCThrow(kInvalidCastException);
-    }
-
-    return(typedByRef.data);
-}
-HCIMPLEND
-
-
 /*************************************************************/
 HCIMPL2(BOOL, JIT_IsInstanceOfException, CORINFO_CLASS_HANDLE type, Object* obj)
```

Comment:
@jkotas, there is a QCall `ThrowInvalidCastException` below this one which uses `COMPlusThrowInvalidCastException`. Should we:
* keep it; in favor of asm helpers https://github.com/dotnet/runtime/pull/108480#issuecomment-2388670105
* return the intermediate result from QCall and throw the actual exception from managed
* pass a callback from managed to QCall `(sourceType, targetType) => throw new InvalidCastExcep..` like some existing GC config QCalls etc. have

Reply:
I think `ThrowInvalidCastException` QCall is fine as is. I do not see what we would gain by "return the intermediate result" or "pass a callback from managed" refactoring. It would just make the code more complicated.

> in favor of asm helpers 

I am not sure what you mean by this. This QCall is not related to any asm helpers.

Reply:
> his QCall is not related to any asm helpers.

Ah, sorry I thought COMPlusThrow ends up using C++ exception and confusing it with your comment in earlier PR https://github.com/dotnet/runtime/pull/109087#discussion_r1811170859.